.github/workflows/release-branches.yml

```diff
@@ -0,0 +1,48 @@
+name: Release Branch
+
+on:
+  push:
+    branches:
+      # NOTE: make sure any branches here are also valid directory names,
+      # otherwise creating the directory and uploading to s3 will fail
+      - 'main'
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # In order to request a JWT for AWS auth
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Download Buildkite Artifacts
+        uses: EnricoMi/download-buildkite-artifact-action@v1.14
+        with:
+          buildkite_token: ${{ secrets.BUILDKITE_TOKEN }}
+          output_path: artifacts
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_S3_UPLOAD_ROLE }}
+          aws-region: us-east-2
+      - name: Publish Release (Branch)
+        run: |
+          sudo chown $USER: -R artifacts/
+
+          LATEST_BRANCH="latest_${{ github.ref_name }}"
+          mkdir "$LATEST_BRANCH"
+          mkdir "$GITHUB_SHA"
+
+          cp nix-installer.sh "$GITHUB_SHA"/
```

Comment:
Something we'll want to fix after this gets in, is that the `nix-installer.sh` script has the "latest release" hardcoded -- arguably we should replace it to point at the proper commit artifact.

We could achieve that by doing something like:

```
sed -i "s@https://install.determinate.systems/nix@https://install.determinate.systems/nix/commit/$GITHUB_SHA@" nix-installer.sh

```

(assuming we have an endpoint for `/nix/commit/:sha`)

Alternatively, we could make a "template" script that will allow us to use something more targeted, like replacing `##__INSTALL_URL__##` with the desired URL prior to releasing / uploading (via GitHub and S3)?

Reply:
Hmmm that URL is from `$NIX_INSTALLER_BINARY_ROOT` so perhaps `curl -L https://install.determinate.systems/nix/rev/boop | NIX_INSTALLER_BINARY_ROOT=https://install.determinate.systems/nix/rev/boop sh -s -- install` will be sufficient

Reply:
Sure, that would work, but specifying the URL twice is kinda ugly and not-nice.

I guess I don't have a strong opinion either way (and it shouldn't affect this merging sooner rather than later), but I'd argue in favor of making it do the right thing, rather than making the user figure out how to do that thing.

Reply:
Why don't we do the sed thing you suggested for now? It seems good.

Reply:
OK, done -- note that the workflows differ slightly in that the branch releaser makes the script use the git revision, while the tag releaser uses the tag. We may want to change this in the future, but I think it should be fine for now.